File: Desktop/SummerVacation/0704/src/numThirtyNine.java
import java.util.ArrayList;
import java.util.List;

class Human{
    private String firstName, secondName,position, date;
    public Human(){
        firstName = null;
        secondName = null;
        position = null;
        date = null;
    }
    public Human setHuman(String fiName, String seName, String job, int day){
        firstName = fiName;
        secondName = seName;
        position = job;
        if(day != 0){
            date = String.valueOf(day/10000)+"-";
            day = day % 10000;
            date = date + String.valueOf(day/100)+"-";
            day = day % 100;
            date = date + String.valueOf(day);
        }
        return this;
    }
    public void showHuman(){
        if(date != null)
        System.out.print(firstName + " " +secondName+"  |"+position+"       |"+date + "\n");
        else
            System.out.print(firstName + " " +secondName+"  |"+position+"       |\n");
    }
    public String getSecondName(){
        return secondName;
    }
}
public class numThirtyNine {
    public List sorting(List<Human> HumanList){
        Human temp = null;
        for(int count =0; count < HumanList.size();count++){
            for(int another = 0; another < HumanList.size(); another++){
                int compare = HumanList.get(another).getSecondName().compareTo(HumanList.get(count).getSecondName());
                if(compare > 0) {
                    temp = HumanList.get(another);
                    HumanList.set(another, HumanList.get(count));
                    HumanList.set(count, temp);
                }
            }
        }
        return HumanList;
    }

    public static void main(String[] args) {
        numThirtyNine example = new numThirtyNine();
        List<Human> humanList= new ArrayList<>();
        humanList.add(new Human().setHuman("John      ","Johnson   ","Manager          ",20161231));
        humanList.add(new Human().setHuman("Tou       ","Xiong     ","Software Engineer",20161015));
        humanList.add(new Human().setHuman("Michaela  ","Michaelson","District Manager ",20151219));
        humanList.add(new Human().setHuman("Jake      ","Jacobson  ","Programmer       ",0));
        humanList.add(new Human().setHuman("Jacpquelyn","Jackson   ","DBA              ",0));
        humanList.add(new Human().setHuman("Sally     ","Weber     ","Web Developer    ",20151218));
        humanList = example.sorting(humanList);
        for(Human a : humanList){
            a.showHuman();
        }
    }
}

File: Desktop/SummerVacation/0704/python/num_thirty_nine.py
class Human :
    def __init__(self,data):
        data = data.split(",")
        self.first_name = data[0]
        self.last_name = data[1]
        self.position = data[2]
        self.date = data[3]
        self.next = None
        self.before = None
    def sort(self, data):
        new_Human = Human(data)
        if new_Human.last_name > self.last_name :
            if self.next is None :
                new_Human.before = self
                self.next = new_Human
            else :
                if self.next.last_name > new_Human.last_name :
                    new_Human.next = self.next
                    new_Human.before = self
                    self.next.before =new_Human
                    self.next = new_Human
                else :
                    Human.sort(self.next,data)
        else :
            if self.before is None :
                self.before = new_Human
                new_Human.next = self
            else :
                if self.before.last_name < new_Human.last_name :
                    new_Human.before = self.before
                    self.before.after = new_Human
                    new_Human.after = self
                    self.before = new_Human
                else :
                    Human.sort(self.before,data)
    def show(self):
        while self.before is not None :
            self = self.before
        while self.next is not None :
            print(self.first_name +"\t"+ self.last_name +"\t"+ self.position +"\t" + self.date)
            self = self.next
        print(self.first_name + "\t" + self.last_name + "\t" + self.position + "\t" + self.date)
john = Human("John,Johnson,Manger,2016-12-31")
john.sort("Tou,Xiong,Software Engineer,2016-10-15")
john.sort("Michaela,Michaelson,District Manager,2015-12-19")
john.sort("Jake,Jacobson,Programmer, ")
john.sort("Jacquelyn,Jackson,DBA, ")
john.sort("Sally,Weber,Web Developer,2015-12-18")
john.show()

